drivers/volume/portworx/portworx.go

```diff
@@ -2736,6 +2768,40 @@ func (d *portworx) getPxctlStatus(n node.Node) (string, error) {
 	return api.Status_STATUS_NONE.String(), nil
 }
 
+func (d *portworx) upgradePortworxUsingOperator(stc *operatorcorev1.StorageCluster, url string, endpointVersion string) error {
+	imageTag := endpointVersion
+	if strings.Contains(url, "edge") {
+		url = fmt.Sprintf("%s/%s/version", url, endpointVersion)
+		stdout, err := exec.Command("curl", "-L", url).CombinedOutput()
+		if err != nil {
+			return fmt.Errorf("error while getting px version url %s, error %v", url, err)
+		}
+		var data interface{}
+		err = yaml.Unmarshal(stdout, &data)
+		if err != nil {
+			return fmt.Errorf("failed to unmarshal px version. cause: %v", err)
+		}
+		versionMap := data.(map[string]interface{})
+		imageTag = versionMap["version"].(string)
+	}
+	stcImage := strings.Split(stc.Spec.Image, ":")[0]
+	stc.Spec.Image = fmt.Sprintf("%s:%s", stcImage, imageTag)
```

Comment:
Just changing `stc.Spec.Image` isn't going to be enough for upgrading, you also need to change `PX_RELEASE_MANIFEST_URL` if we are using `edge-install`